### DISCUSSION TITLE
How to add a custom OAuth service?

### DISCUSSION THREAD

**Discussion by livingsilver94** [oss-apps/split-pro#86] [Q&A] [answered]
I see that NextAuth supports [custom OAuth providers](https://next-auth.js.org/configuration/providers/oauth#using-a-custom-provider) but SplitPro only offers Google. I'd like to use Authelia for my self-hosted SplitPro instance.

NextAuth documents how to register a custom provider pretty well but I don't know how to integrate the snippet the documentation provides in SplitPro. Help would be appreciated :)

**Comment by KMKoushik** [comment, marked as answer]
you can add the provider [here](https://github.com/oss-apps/split-pro/blob/main/src/server/auth.ts#L55), but should be enabled based on the env variables.

Here is an example for conditional adds https://github.com/unsend-dev/unsend/blob/main/apps/web/src/server/auth.ts#L55

**Reply by KMKoushik** [reply]
lemme know if it make sense
